### PR TITLE
Fix some todos

### PR DESCRIPTION
**Pull request by mattwynne**
### 🤔 What's changed?

* Factored out the environment / argument integrity guards into functions
* Guard for latest git tags not being fetched locally (the "already released" check would be unreliable otherwise)
* Quiet output from pushd/popd/git/npm
* Check for missing tools in environment
* Show unreleased changes from `CHANGELOG` if user omits release number

### ⚡️ What's your motivation? 

Progress!

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by mattwynne** [comment]
I managed to test the display of unreleased changes, but I think other changes here will be too tough to test, probably (e.g. missing tools), or not worth it (e.g. quiet output)

**Review by mpkorstanje — CHANGES_REQUESTED** [review]

File: release [hunk 1]
```diff
@@ -41,31 +57,65 @@ function release_ruby() {
...
-# Version4:
-# Bootstrap from single git repo.
-# Add tests
+  if [ -n "$(git tag --list "v$NEW_VERSION")" ]; then
+    echo "Version $NEW_VERSION has already been released."
+    exit 1
+  fi
+}
+
+function validate_environment() {
```
[the remark this answers is not in the document]
Comment (mattwynne) [reply release]: fixed in 3404596e2191f29702020ed90918a67e91365ee2

File: release [hunk 2]
```diff
@@ -41,31 +57,65 @@ function release_ruby() {
...
+
+   required_tools=("git" "changelog")
+   [[ -d javascript ]] && required_tools+=("npm")
+   [[ -d java ]] && required_tools+=("mvn")
```
[the remark this answers is not in the document]
Comment (mattwynne) [reply release]: Good point. I've fixed it by making the tool-checking parameterised so we can do it from anywhere we have a dependency on a tool.

See 3404596e2191f29702020ed90918a67e91365ee2

WDYT?

**Review by mpkorstanje — APPROVED** [review]
LGTM. Lets keep this train going.

**Pull request merged**